pypolychord/output.py

```diff
@@ -141,8 +155,108 @@ def make_paramnames_files(self, paramnames):
         for i, _ in enumerate(self.logZs):
             self.make_paramnames_file(paramnames,
                                       self.cluster_paramnames_file(i))
-
+        self._create_pandas_table(paramnames = paramnames)
+            
+            
     def make_paramnames_file(self, paramnames, filename):
         with open(filename, 'w') as f:
             for name, latex in paramnames:
                 f.write('%s   %s\n' % (name, latex))
+
+    def _create_pandas_table(self, paramnames = None):
+
+        initial_col_names = ['weight','loglike']
+
+        n_params = np.genfromtxt('%s_equal_weights.txt' % self.root).shape[1] - 2
+
+        # build the paranames for the table
+        if paramnames is None:
+            for i in range(n_params):
+
+                initial_col_names.append('p%d'%i)
+        else:
+
+            initial_col_names.extend(paramnames)
+        
+    
+
+        # now read the table
+
+        self._samples_table = pd.read_table('%s_equal_weights.txt' % self.root,sep=' ',
+                                            skipinitialspace=1,
+                                            names= initial_col_names)
+
+        # correct to loglike
+        self._samples_table['loglike'] *= -0.5 
+
+    def display(self):
```

Comment:
It would be best if there were a `__str__(self)` function that returned a string here, and then have `display` simply return this string. This would allow one to get the same output if one called `print(samples)`, which is more pythonic than `print(samples.display())`